python/ray/autoscaler/autoscaler.py

```diff
@@ -824,3 +854,22 @@ def add_hash_of_file(fpath):
         _hash_cache[conf_str] = hasher.hexdigest()
 
     return _hash_cache[conf_str]
+
+def request_cores(redis_address, cores, redis_password=None):
+    """Remotely request some CPU cores from the autoscaler.
+    This function is to be called e.g. on a node before submitting a bunch of
+    ray.remote calls to ensure that resources rapidly become available.
+
+    In the future this could be extended to do GPU cores or other custom
+    resources.
+
+    This function is non blocking.
+
+    Arguments:
+
+    redis_address: str      -- the redis address of the head ray node (required)
```

Comment:
these should all be indented by 4 spaces

Reply:
Done